src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -1100,23 +1149,33 @@ class ContentSidebar extends PureComponent<Props, State> {
             }
         };
 
-        let tasks = tasksWithoutAssignments;
         const { entries } = tasksWithoutAssignments;
-        const taskAssignmentPromises = [];
-        entries.forEach((task) => {
-            const promise = this.api.getTasksAPI(shouldDestroy).getAssignments(
-                fileId,
-                task.id,
-                (assignments) => {
-                    tasks = this.fetchTaskAssignmentsSuccessCallback(tasks, assignments);
-                },
-                this.fetchTasksErrorCallback,
-                requestData
-            );
-            taskAssignmentPromises.push(promise);
+        const formattedTasks = { total_count: 0, entries: [] };
+        const assignmentPromises = [];
+        entries.forEach((task: Task) => {
+            const assignmentPromise = new Promise((resolve) => {
+                this.api.getTasksAPI(shouldDestroy).getAssignments(
+                    fileId,
+                    task.id,
+                    (assignments) => {
+                        formattedTasks.entries.push(this.appendAssignmentsToTask(task, assignments.entries));
+                        formattedTasks.total_count += 1;
+                        resolve();
+                    },
+                    (e) => {
+                        this.fetchTasksErrorCallback(e);
+                        resolve();
```

Comment:
We resolve either way because we still want to show the valid tasks even if a few assignment fetches fail